### ISSUE TITLE
NVM for windows is failing when using npm prestart scripts

### ISSUE THREAD

**Issue by Dreier14** [coreybutler/nvm-windows#835]
**Issue:**
When I try to run my project, doing `npm start` I am getting the below. We have this chain in our project since some repos are using different node versions. I have nvm-windows 1.1.9 latest installed. I have tried a machine recycle, re-installing nvm for windows, deleting the packages and modules and much more. Whats interesting is that when I remove the prestart script and am on the required version by doing `nvm use 18.4.0` this works, also when I add back in the `prestart `script and use for instance `nvm use 12.22.12` the prestart script executes fine when running `npm start` and the version changes to 18.4.0 with no issue, but if I stop the project and start it again it fails with the below and either I need to remove the prestart script or change to a different version. I am not understanding how this could be. **I AM IN NO WAY USING THE LINUX/MAC Version of NVM**

**How To Reproduce:**
Update nvm-windows to latest version,
add prestart script in package.json,
prestart: nvm install 18.4.0 && nvm use 18.4.0,
use the same version your project needs and re run the npm start

Result:
`prestart: nvm install 18.4.0 && nvm use 18.4.0` //npm script in pckage.json

```
nvm install 18.4.0 && nvm use 18.4.0

This is not the package you are looking for: please go to http://nvm.sh/
```

**Expected Behavior:**
There should be no error and the project should start automatically.

**Screenshots:**
![image](https://user-images.githubusercontent.com/37100632/180050225-f07fa70a-059d-4912-a9e4-3645be19ec71.png)
![image](https://user-images.githubusercontent.com/37100632/180050130-3d937d0f-179e-4988-a95e-49be5e8f3a10.png)
![image](https://user-images.githubusercontent.com/37100632/180050351-bfa5fc4e-dba4-4936-a50b-995ae1952682.png)

**Desktop:**
Windows Platform

**Comment by coreybutler** [comment]
npm is looking for a local executable within node_modules, not the `%PATH%`. To confirm, specify the absolute path to nvm.exe in your npm script (example: C:\nvm\nvm.exe install 18.4.0 && C:\nvm\nvm.exe use 18.4.0`).

The output suggests you're using _something_ POSIX based, even if you're on Windows... such as WSL or a special shell (like bash for windows). The tip-off is the terminal prompt. Normally it looks like:

**CMD**
<img width="320" alt="image" src="https://user-images.githubusercontent.com/770982/180075822-f701e2f2-98d6-4489-849a-f873a22caa97.png">

**PowerShell**
<img width="183" alt="image" src="https://user-images.githubusercontent.com/770982/180075896-1adb22ad-c39b-42c9-bb7b-eda8174f94ea.png">

_It does not normally look like this (unless you explicitly set a custom `$` prompt):_
<img width="38" alt="image" src="https://user-images.githubusercontent.com/770982/180076061-e57c53fb-811c-4b42-aa06-f510281984ae.png">

For info about shell support, see https://github.com/coreybutler/nvm-windows/wiki/Common-Issues#special-terminal-shells.

Can you clarify which shell you're using?

**Comment by Dreier14** [comment]
![image](https://user-images.githubusercontent.com/37100632/180096815-0b2a235e-b147-4ec2-83c8-3595d2ca1436.png)
![image](https://user-images.githubusercontent.com/37100632/180097447-eb4a6a34-adb9-4944-b9b3-6002ef7efdaf.png)

Strangely enough if I am using 12.22.12 node version and run the npm start script with the prestart there is no issue. Both CMD and Windows PS below. **Please note that I am not using the absolute path solution for this to work.

![image](https://user-images.githubusercontent.com/37100632/180097739-f699d0b2-1467-4f3a-a58a-180774f67a6f.png)
![image](https://user-images.githubusercontent.com/37100632/180098087-fc724d36-ee49-4cc7-a96b-8784f0a8aabb.png)


My mistake for posting Git Bash, I was trying a different shell as both PS Admin and CMD were giving me these issues first. What's interesting is my ENV variables are all correct and pointing to the PATH's that my teammates are also using and one person on my team is using git bash with no side effects. Only when I add example: C:\nvm\nvm.exe install 18.4.0 && C:\nvm\nvm.exe use 18.4.0` is it not giving me any issues, but ideally we would like to avoid this in our Azure Pipeline as we have been using the above ```prestart: nvm install 18.4.0 && nvm use 18.4.0``` with no issues. Still yet I haven't been able to find any resolution except for the absolute path approach. I had even double checked my teammates ENV's which also appear to be the same and the location where nvm for windows is installed

**Comment by coreybutler** [comment]
I ran a quick test and I think the issue has to do with the environment, not NVM for Windows. Here's a simple test to confirm:

**package.json**
```javascript
{
  "name": "tmp",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "start": "node index.js",
    "prestart": "nvm install latest && nvm use latest",
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "",
  "license": "ISC"
}
```

**index.js**
```javascript
console.log(`${process.platform} ${process.arch} ${process.version}`)
```

**output**

<img width="652" alt="image" src="https://user-images.githubusercontent.com/770982/180099657-ebea3a98-c6b4-4543-a462-8cdc8140e362.png">

--

Between the test I've run above in a clean Windows environment and the fact that absolute paths work in your environment is a very strong indicator that that the problem is in the environment. This is really more of an issue using npm and how npm resolves executables.

Bottom line: I don't think there is anything NVM for Windows can do that would resolve this. I'm not sure why it is showing "please go to http://nvm.sh", because that's a completely different project that has to be installed separately (and it does not work in Windows). It suggests something in the npm environment thinks nvm.sh should exist.

Seeing as this doesn't appear to be an issue with NVM for Windows, I'm going to close the issue. I hope this test case helps you track down the answer.

**Comment by Dreier14** [comment]
After pretty much nuking everything on my VDI I finally got this to work, want to Thank you Corey for your prompt response. If I can be any assistance on this project in the future with PR's please reach out :)

**Comment by Dreier14** [comment]
You are welcome! Submitted that feedback. Also interestingly I did some testing to see if this works with git bash and have not come into any issues. Not sure what other users may have faced but looks as if this is compatible.
